sel4-extract/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+# set minimum cmake verision
+cmake_minimum_required(VERSION 3.0.0)
+
+# set project name
+project(sel4-extract)
+
+# add executable
+add_executable (sel4-extract sel4-extract.c)
+
+# set compiler options
+set(FLAGS_WAR_AS_ERR -Wall -Wpedantic -Werror)
+# set(FLAGS_MEM_DBG -fsanitize=address -fsanitize=leak -fsanitize=bounds)
+
+target_compile_options(sel4-extract PRIVATE ${FLAGS_WAR_AS_ERR} ${FLAGS_MEM_DBG})
+target_link_options(sel4-extract  PRIVATE ${FLAGS_WAR_AS_ERR} ${FLAGS_MEM_DBG})
```

Comment:
Treat compiler and linker options separately, don't mush them up (i.e. own variables for preprocessor, compiler and linker flags. By convention these are called CPPFLAGS, CFLAGS and LDFLAGS respectively.